lib/src/package_analysis/shape.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'shape.g.dart';
+
+@JsonSerializable()
+class PackageShape {
+  final List<LibraryShape> libraries;
+  final List<ClassShape> classes;
+
+  PackageShape(this.libraries, this.classes);
+
+  factory PackageShape.fromJson(Map<String, dynamic> json) =>
+      _$PackageShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$PackageShapeToJson(this);
+}
+
+@JsonSerializable()
+class LibraryShape {
+  final String identifier;
+  @JsonKey(name: 'exported-classes')
+  final Set<int> exportedClasses;
```

Comment:
I would suggest documenting this property too... something saying that this is `[ClassShape.id]` of the classes that are exported.